File: Examples/Simple Sync/Simple Sync/NumberHolder.m
//
//  NumberHolder.m
//  Magical Record
//
//  Created by Drew McCormack on 18/04/14.
//  Copyright (c) 2014 Drew McCormack. All rights reserved.
//

#import "NumberHolder.h"


@implementation NumberHolder

@dynamic uniqueIdentifier;
@dynamic number;

+ (instancetype)numberHolderInManagedObjectContext:(NSManagedObjectContext *)context
{
    NSFetchRequest *fetch = [NSFetchRequest fetchRequestWithEntityName:@"NumberHolder"];
    NumberHolder *holder = [[context executeFetchRequest:fetch error:NULL] lastObject];
    if (!holder) {
        holder = [NSEntityDescription insertNewObjectForEntityForName:@"NumberHolder" inManagedObjectContext:context];
    }
    return holder;
}

@end

File: Examples/Simple Sync in Swift/Simple Sync in Swift/NumberHolder.swift
//
//  NumberHolder.swift
//  Simple Sync in Swift
//
//  Created by Drew McCormack on 31/01/16.
//  Copyright © 2016 Drew McCormack. All rights reserved.
//

import UIKit
import CoreData

class NumberHolder: NSManagedObject {
    
    @NSManaged var uniqueIdentifier: NSString
    @NSManaged var number: NSNumber
    
    class func numberHolderInContext(_ context:NSManagedObjectContext) -> NumberHolder {
        var holder:NumberHolder?
        context.performAndWait {
            let fetch = NSFetchRequest<NumberHolder>(entityName: "NumberHolder")
            holder = try! fetch.execute().last
            if holder == nil {
                holder = NumberHolder(context:context)
            }
        }
        return holder!
    }

}
